book/src/common_patterns/selection.md

```diff
@@ -0,0 +1,79 @@
+# Selection
+
+The "selection" (or "firewall") pattern is when you have a query Qsel that reads from some
+other Qbase and extracts some small bit of information from Qbase that it returns.
+In particular, Qsel does not combine values from other queries. In some sense,
+then, Qsel is redundant -- you could have just extracted the information
+the information from Qbase yourself, and done without the salsa machinery. But
+Qsel serves a role in that it limits the amount of re-execution that is required
+when Qbase changes.
+
+## Example: the base query
+
+For example, imagine that you have a query `parse` that parses the input text of a request to return a
+only consume small parts of itheader and a body:
```

Comment:
```suggestion
only consume small parts of it (a header and a body):
```

Reply:
man emacs was doing weird stuff like this whenever I tried to "reflow" paragraphs :( I thought I caught them all...